core/src/main/scala/org/apache/spark/internal/config/package.scala

```diff
@@ -1217,6 +1217,14 @@ package object config {
       .stringConf
       .createWithDefault(classOf[LocalDiskShuffleDataIO].getName)
 
+  private[spark] val SHUFFLE_LOCATION_PLUGIN_CLASS =
+    ConfigBuilder("spark.shuffle.sort.location.plugin.class")
+      .doc("Qualified name of the class that used to initiate plugin location instance. " +
+        "If not specified, Spark will use its native location (a.k.a BlockManagerId) by default.")
+      .version("3.2.0")
+      .stringConf
+      .createOptional
```

Comment:
Do we need to use `createOptional` here? Instead of saying `"If not specified, Spark will use its native location (a.k.a BlockManagerId) by default.")`, can we specify the class here explicitly?

Reply:
I used the `optional` intentionally.

I tried the way of the default value (`BlockManagerId`) previously.  And I realized that we wouldn't expect to load the `BlockManager` using the reflection, which is low efficiency, but create it directly as it's a built-in class in Spark. As a result, the default value won't be used at all (or it would be only used to see if it's not the default value by comparison, but this's meaningless to do so).